pubtools/_quay/utils/misc.py

```diff
@@ -284,3 +285,35 @@ def fn_wrapper(*args, **kwargs):
         return fn_wrapper
 
     return decorate
+
+
+def get_pyxis_ssl_paths(target_settings):
+    """
+    Get certificate and key paths for Pyxis SSL authentication.
+
+    First attempt is made by invoking the hook implementation 'get_cert_key_paths_plugin'.
+    If nothing is returned (no hook implementation is registered), fallback on target settings
+    values of 'pyxis_ssl_cert' and 'pyxis_ssl_key'. If multiple values are returned (multiple
+    hook implementations are registered), take the first response (from the implementation which
+    was registered last). Otherwise, raise an error.
+
+    Args:
+        target_settings (dict):
+            Dictionary containing various task-related settings.
+    Returns ((str, str)):
+        Paths to Pyxis SSL certificate and key.
+    """
+    result = pm.hook.get_cert_key_paths_plugin(server_url=target_settings["pyxis_server"])
+    if result:
```

Comment:
This will use the first result, but I think you should instead use the first non-None result.  Reason for this is that it allows hooks to indicate "sorry, I don't know what to do here" - e.g. a hook which knows how to get cert/key for some services and not others - and then let another hook take over.

This brings me back to 'firstresult' again because https://pluggy.readthedocs.io/en/latest/#collecting-results says that's exactly what happens if you use firstresult on the hookspec.